SingularityService/src/main/java/com/hubspot/singularity/hooks/LoadBalancerClient.java

```diff
@@ -18,4 +21,10 @@
   SingularityLoadBalancerUpdate cancel(LoadBalancerRequestId loadBalancerRequestId);
 
   SingularityLoadBalancerUpdate delete(LoadBalancerRequestId loadBalancerRequestId, String requestId, Set<String> loadBalancerGroups, String serviceBasePath);
+
+  Collection<UpstreamInfo> getBaragonUpstreamsForRequest(String requestId);
```

Comment:
nit on naming. The original intent of the interface here was that we could eventually implement a non-baragon-specific one if needed. While it doesn't seem like that will actually happen, the more correct naming here would remove baragon from the method name (more like just `getUpstreamsForRequest`). Similar for the`makeAndSendBaragonRequest`